extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultPKIManager.java

```diff
@@ -0,0 +1,615 @@
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKICRLRotateResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKICertificateListResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKICertificateResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIConfigCABody;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIConfigCRLBody;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIConfigCRLData;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIConfigCRLResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIConfigURLsBody;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIConfigURLsData;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIConfigURLsResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIGenerateCertificateBody;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIGenerateCertificateData;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIGenerateCertificateResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIGenerateIntermediateCSRBody;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIGenerateIntermediateCSRData;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIGenerateIntermediateCSRResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIGenerateRootBody;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIGenerateRootData;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIGenerateRootResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIRevokeCertificateBody;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIRevokeCertificateResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIRoleOptionsData;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIRoleReadResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIRoleUpdateBody;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKIRolesListResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKISetSignedIntermediateCABody;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKISignCertificateRequestBody;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKISignCertificateRequestData;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKISignCertificateRequestResult;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKISignIntermediateCABody;
+import io.quarkus.vault.runtime.client.dto.pki.VaultPKITidyBody;
+import io.quarkus.vault.runtime.client.secretengine.VaultInternalPKISecretEngine;
+
+@ApplicationScoped
+public class VaultPKIManager implements VaultPKISecretEngine {
+
+    private final String mount;
+    private final VaultAuthManager vaultAuthManager;
+    private final VaultInternalPKISecretEngine vaultInternalPKISecretEngine;
+
+    @Inject
+    public VaultPKIManager(
+            VaultAuthManager vaultAuthManager,
+            VaultInternalPKISecretEngine vaultInternalPKISecretEngine) {
+        this("pki", vaultAuthManager, vaultInternalPKISecretEngine);
+    }
+
+    VaultPKIManager(
```

Comment:
Perfectly open to other ways for other usages but our usage requires the ability to check for and enable/disable PKI engines dynamically.  This is the only way I could figure out how to implement entirely dynamic PKI engines.

FYI, as mentioned elsewhere, listing them in a configuration is not something we could do.

Reply:
> FYI, as mentioned elsewhere, listing them in a configuration is not something we could do.

yes. perfectly understood. may be @mkouba can you give us some advice.

basically there is a class [VaultPKIManager](https://github.com/kdubb/quarkus/blob/feature/vault_pki/extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultPKIManager.java#L70-L75) that is used sometimes as a cdi bean and sometimes as a [regular class](https://github.com/kdubb/quarkus/blob/feature/vault_pki/extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultPKIManager.java#L77-L84), instantiated by another CDI bean [VaultPKIManagerFactory](https://github.com/kdubb/quarkus/blob/feature/vault_pki/extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultPKIManagerFactory.java#L28-L38), because we want the name (i.e. technically the vault mount path) to by dynamic.

is there a better way to implement this pattern?

Reply:
I'm sorry but I have no idea. If need pls try to prepare a generalized example and send a question to the quarkus-dev...

Reply:
as suggested: https://groups.google.com/g/quarkus-dev/c/yaFSirAG_DA
@kdubb would you be able to add more context as to why you cannot list the PKIs in your configuration?

Reply:
I do not think we will have a better solution. so let's keep it that way.